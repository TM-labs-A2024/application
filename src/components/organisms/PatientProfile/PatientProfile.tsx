import { Text, Heading, Stack, Link } from '@chakra-ui/react'
import { Patient as PatientType } from '@src/types'
import { isIOS } from '@utils/index'
import { format } from 'date-fns'
import { es } from 'date-fns/locale/es'
import NextLink from 'next/link'
import React from 'react'

export default function PatientProfile({ patient }: { patient: PatientType }) {
  return (
    <div
      className={`mx-auto block h-screen w-screen overflow-hidden px-8 lg:px-96 ${isIOS() ? 'pt-20' : 'pt-8'}`}
      data-testid="patient-profile"
    >
      <div className="mb-8 flex w-full flex-row justify-between">
        <Heading as="h2" size="md" noOfLines={1}>
          Perfil
        </Heading>
        <Link as={NextLink} href="/perfil/editar">
          Editar
        </Link>
      </div>
      <Stack spacing={1} mb={6}>
        <Heading as="h3" size="md" noOfLines={1}>
          {patient.firstname} {patient.lastname}
        </Heading>
        <Text>CI: {patient.govId.toLocaleString()}</Text>
      </Stack>
      <Stack mb={6}>
        <h4 className="text-sm text-gray-600">Fecha de nacimiento</h4>
        <Text className="font-medium">
          {format(new Date(patient.birthdate), "dd 'de' MMMM, yyyy", {
            locale: es
          })}
        </Text>
      </Stack>
      <Stack mb={6}>
        <h4 className="text-sm text-gray-600">Correo electrónico</h4>
        <Text className="font-medium">{patient.email}</Text>
      </Stack>
      <Stack mb={6}>
        <h4 className="text-sm text-gray-600">Teléfono</h4>
        <Text className="font-medium">{patient.phoneNumber}</Text>
      </Stack>
    </div>
  )
}
