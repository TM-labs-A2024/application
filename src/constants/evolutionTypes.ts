export const evolutionTypes = [
  {
    id: '1',
    name: 'Evolución'
  },
  {
    id: '2',
    name: 'Hospitalización'
  },
  {
    id: '3',
    name: 'Alta'
  },
  {
    id: '4',
    name: 'Primera consulta'
  }
]
