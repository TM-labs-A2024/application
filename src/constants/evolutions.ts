export const evolution = {
  specialty: {
    description: '',
    id: '',
    name: ''
  },
  author: '',
  title: '',
  description: '',
  'content-format': '',
  created_at: '',
  payload: {
    type: {
      value: '',
      label: ''
    },
    reason: '',
    description: '',
    history: '',
    examination: '',
    summary: '',
    diagnostic: '',
    comments: ''
  }
}
