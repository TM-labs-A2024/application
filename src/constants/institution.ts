export const institution = {
  id: '3fa85f64-5717-4562-b3fc-2c963f66afa1',
  govId: '123456',
  name: 'Clínica Tex Mex',
  credentials: '14578245',
  address: 'Av. Fernandez Peña, 147.',
  email: 'test@test.com',
  phoneNumber: '+582742214441',
  institutionUser: {
    id: '',
    firstname: '',
    lastname: '',
    govId: '',
    birthdate: '',
    email: 'test@test.com',
    phoneNumber: '+582742214441',
    role: '',
    institutionId: ''
  }
}
