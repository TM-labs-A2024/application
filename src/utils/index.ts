import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
