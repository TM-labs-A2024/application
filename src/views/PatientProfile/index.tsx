import { PatientsLayout } from '@layouts/index'
import PatientProfile from '@src/components/organisms/PatientProfile'
import { Patient as PatientType } from '@src/types'
import React from 'react'

export default function PatientProfileView({
  patient,
  onLogout
}: {
  patient: PatientType
  onLogout?: () => void
}) {
  return (
    <PatientsLayout>
      <PatientProfile patient={patient} onLogout={onLogout} />
    </PatientsLayout>
  )
}
