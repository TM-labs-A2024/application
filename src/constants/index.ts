import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
