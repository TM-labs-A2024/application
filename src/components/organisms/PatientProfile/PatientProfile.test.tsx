import { patient } from '@src/constants'
import { render, screen } from '@test/utils/index'
import { format } from 'date-fns'
import { es } from 'date-fns/locale/es'

import '@testing-library/jest-dom'
import PatientProfile from './PatientProfile'

describe('Organisms > Patient test', () => {
  test('The component shows the name, govId, birthdate, email and phoneNumber of the patient', () => {
    render(<PatientProfile patient={patient} />)

    const idMatcher = (content: string) => content.includes(patient.govId)
    const patientId = screen.getByText(idMatcher)

    const emailMatcher = (content: string) => content.includes(patient.email)
    const patientEmail = screen.getByText(emailMatcher)

    expect(screen.getByText(`${patient.firstname} ${patient.lastname}`)).toBeInTheDocument()
    expect(patientId).toBeInTheDocument()
    expect(
      screen.getByText(
        format(new Date(patient.birthdate), "dd 'de' MMMM, yyyy", {
          locale: es
        })
      )
    ).toBeInTheDocument()
    expect(patientEmail).toBeInTheDocument()
    expect(screen.getByText(patient.phoneNumber)).toBeInTheDocument()
  })

  test('The component allows to go to the patients details edition form', () => {
    render(<PatientProfile patient={patient} />)

    expect(screen.getByText('Editar')).toHaveAttribute('href', '/perfil/editar')
  })

  test('Matches the snapshot', () => {
    render(<PatientProfile patient={patient} />)

    expect(screen.getByTestId('patient-profile')).toMatchSnapshot()
  })
})
