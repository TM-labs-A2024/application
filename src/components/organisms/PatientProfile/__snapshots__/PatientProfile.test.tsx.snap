// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > Patient test Matches the snapshot 1`] = `
<div
  class="mx-auto block h-screen w-screen overflow-hidden px-8 lg:px-96 pt-8"
  data-testid="patient-profile"
>
  <div
    class="mb-8 flex w-full flex-row justify-between"
  >
    <h2
      class="chakra-heading css-ypohwm"
    >
      Perfil
    </h2>
    <a
      class="chakra-link css-spn4bz"
      href="/perfil/editar"
    >
      Editar
    </a>
  </div>
  <div
    class="chakra-stack css-1uriqkk"
  >
    <h3
      class="chakra-heading css-ypohwm"
    >
      José Feliciano
       
      Gutierrez García
    </h3>
    <p
      class="chakra-text css-0"
    >
      CI: 
      E99885561
    </p>
  </div>
  <div
    class="chakra-stack css-1j35777"
  >
    <h4
      class="text-sm text-gray-600"
    >
      Fecha de nacimiento
    </h4>
    <p
      class="chakra-text font-medium css-0"
    >
      11 de noviembre, 1999
    </p>
  </div>
  <div
    class="chakra-stack css-1j35777"
  >
    <h4
      class="text-sm text-gray-600"
    >
      Correo electrónico
    </h4>
    <p
      class="chakra-text font-medium css-0"
    >
      user@example.com
    </p>
  </div>
  <div
    class="chakra-stack css-1j35777"
  >
    <h4
      class="text-sm text-gray-600"
    >
      Teléfono
    </h4>
    <p
      class="chakra-text font-medium css-0"
    >
      +584127659814
    </p>
  </div>
  <button
    class="chakra-button mt-56 w-full css-jzcxx5"
    type="button"
  >
    Cerrar sesión
  </button>
</div>
`
